tldw_Server_API/app/core/Explainer/grounding.py

```
@@ -0,0 +1,173 @@
+    return GroundedExplainerChild(
+        title=_text_or_default(child.get("title"), "Generated explanation"),
+        body=str(child.get("body")) if child.get("body") is not None else None,
+        kind=_text_or_default(child.get("kind"), "explanation"),
+        intent=_text_or_default(child.get("intent"), fallback_intent),
+        evidence_state=evidence_state,
+        outside_knowledge_used=outside_knowledge_used,
+        citations=citations,
+    )
+
+
+def _coerce_citations(value: Any) -> list[dict[str, Any]]:
```

Ruff (0.15.15)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)